_drafts/PR-blogs/planning-docs/example-code/animated-dots/original/script.js

```diff
@@ -0,0 +1,170 @@
+window.onload = init();
+function init() {
+  canvas = document.getElementById("canvas");
+  context = canvas.getContext("2d");
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  mouse = { x: 0, y: 0 };
+  colors = [
+    "#af0",
+    "#7CD14E",
+    "#1CDCFF",
+    "#FFFF00",
+    "#FF0000",
+    "#aee137",
+    "#bef202",
+    "#00b2ff",
+    "#7fff24",
+    "#13cd4b",
+    "#c0fa38",
+    "#f0a",
+    "#a0f",
+    "#0af",
+    "#000000",
+  ];
+  canvas.addEventListener("mousemove", MouseMove, false);
+  canvas.addEventListener("mousedown", MouseDown, false);
+  canvas.addEventListener("mouseup", MouseUp, false);
+  window.addEventListener("resize", canvasResize, false);
+  dotsHolder = [];
+  mouseMove = false;
+  mouseDown = false;
+  for (i = 0; i < 1000; i++) {
+    dotsHolder.push(new dots());
+  }
+
+  /*REQUEST ANIMATION FRAME*/
+  (function () {
+    var lastTime = 0;
+    var vendors = ["ms", "moz", "webkit", "o"];
+    for (var x = 0; x < vendors.length && !window.requestAnimationFrame; ++x) {
+      window.requestAnimationFrame =
+        window[vendors[x] + "RequestAnimationFrame"];
+      window.cancelAnimationFrame =
+        window[vendors[x] + "CancelAnimationFrame"] ||
+        window[vendors[x] + "CancelRequestAnimationFrame"];
+    }
+
+    if (!window.requestAnimationFrame)
+      window.requestAnimationFrame = function (callback, element) {
+        var currTime = new Date().getTime();
+        var timeToCall = Math.max(0, 16 - (currTime - lastTime));
+        var id = window.setTimeout(function () {
+          callback(currTime + timeToCall);
+        }, timeToCall);
+        lastTime = currTime + timeToCall;
+        return id;
+      };
+
+    if (!window.cancelAnimationFrame)
+      window.cancelAnimationFrame = function (id) {
+        clearTimeout(id);
+      };
+  })();
+}
+function canvasResize(event) {
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  cancelAnimationFrame(animate);
+}
+function MouseUp(event) {
+  if (mouseMove) {
+    mouseMove = false;
+  }
+  if (mouseDown) {
+    mouseDown = false;
+  }
+}
+function MouseDown(event) {
+  mouseDown = true;
+}
+function MouseMove(event) {
+  mouse.x = event.pageX - canvas.offsetLeft;
+  mouse.y = event.pageY - canvas.offsetTop;
+  if (mouseMove) {
+    context.lineTo(mouseX, mouseY);
+    context.stroke();
+  }
```

Comment:
Putting a little whitespace between `mouse.y = ...` and `if (mouseMove)` will  make this just a little bit more readable. This gets back to the idea of grouping like-things together: updating the mouse x and y positions and drawing a line on the canvas are both things that happen on `mousemove`, but are nevertheless separate actions. Giving a little breathing room between these two actions will aide comprehension for newcomers to the code.